src/bin/ion/commands/generate/templates/java/enum.templ

```diff
@@ -0,0 +1,63 @@
+{% set full_namespace = namespace | join(sep=".") %}
+{% if is_nested == false %}
+package {{ full_namespace }};
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonException;
+import com.amazon.ion.IonWriter;
+import com.amazon.ion.IonType;
+import java.io.IOException;
+{% endif %}
+
+{# Verify that the abstract data type is a enum and store information for this enum #}
+{% set enum_info = model.code_gen_type["Enum"] %}
+
+{% if is_nested == true %} static {% endif %} enum {{ model.name }} {
+    {% for variant in enum_info["variants"] -%}
+           {{ variant.name | upper }} {
+                    @Override
+                    public void writeTo(IonWriter writer) throws IOException {
+                         {# Writes the enum variant as a string or symbol  #}
+                         writer.write{{variant.variant_type}}("{{ variant.name }}");
+                    }
+           }
+           {% if loop.last == false %},{% endif %}
```

Comment:
Enum variants are one of the few places where Java lets you include trailing commas.
```suggestion
           },
```